app/scripts/modules/core/loadBalancer/loadBalancer.write.service.js

```diff
@@ -1,68 +0,0 @@
-'use strict';
-
-import _ from 'lodash';
-import {TASK_EXECUTOR} from 'core/task/taskExecutor';
-
-let angular = require('angular');
-import {INFRASTRUCTURE_CACHE_SERVICE} from 'core/cache/infrastructureCaches.service';
-
-module.exports = angular
-  .module('spinnaker.core.loadBalancer.write.service', [TASK_EXECUTOR, INFRASTRUCTURE_CACHE_SERVICE])
-  .factory('loadBalancerWriter', function(infrastructureCaches, taskExecutor) {
-
-    function deleteLoadBalancer(loadBalancer, application, params = {}) {
-      params.type = 'deleteLoadBalancer';
-      params.loadBalancerName = loadBalancer.name;
-      params.regions = [loadBalancer.region];
-      params.credentials = loadBalancer.accountId;
-      params.cloudProvider = loadBalancer.cloudProvider;
-
-      var operation = taskExecutor.executeTask({
-        job: [params],
-        application: application,
-        description: 'Delete load balancer: ' + loadBalancer.name
-      });
-
-      infrastructureCaches.clearCache('loadBalancers');
-
-      return operation;
-    }
-
-
-    function upsertLoadBalancer(loadBalancer, application, descriptor, params = {}) {
-      var name = loadBalancer.clusterName || loadBalancer.name;
-      let protocol = loadBalancer.healthCheckProtocol || '';
-      if (protocol.startsWith('HTTP')) {
-        loadBalancer.healthCheck = `${protocol}:${loadBalancer.healthCheckPort}${loadBalancer.healthCheckPath}`;
-      } else {
-        loadBalancer.healthCheck = `${protocol}:${loadBalancer.healthCheckPort}`;
-      }
-      loadBalancer.type = 'upsertLoadBalancer';
-      loadBalancer.availabilityZones = {};
-      loadBalancer.availabilityZones[loadBalancer.region] = loadBalancer.regionZones || [];
-      if (!loadBalancer.vpcId && !loadBalancer.subnetType) {
-        loadBalancer.securityGroups = null;
-      }
-
-      // We want to extend params with all attributes from loadBalancer, but only if they don't already exist.
-      _.assignWith(params, loadBalancer, function(value, other) {
```

Comment:
The google network load balancer put the health check on `params` (see: https://github.com/spinnaker/deck/blob/master/app/scripts/modules/google/loadBalancer/configure/network/createLoadBalancer.controller.js#L210-L216), which allowed it to pass through into the upsert description. That can't happen in the new load balancer writer, since you write the health check to the job after you merge the params, load balancer, and job. Does that health check bit have to go in the writer? It feels like an AWS only thing - I'm pretty sure GCE, Kubernetes, and App Engine don't use it.